src/kaboodle.rs

```diff
@@ -742,3 +748,33 @@ impl KaboodleInner {
         }
     }
 }
+
+fn calculate_peer_latency(peer_info: &PeerInfo) -> Option<Duration> {
+    let ping_send_time = match peer_info.state {
```

Comment:
This block of code feels very awkward to me:

- assignment
- a match
- an explicit `return`

I am open to suggestions.

Reply:
Hm. Yeah I'm not a fan either but the solution I just thought of kind of makes it worse so... :-/